src/role.ts

```diff
@@ -146,25 +207,32 @@ export class GithubActionsRole extends iam.Role {
    */
   constructor(scope: Construct, id: string, props: GithubActionsRoleProps) {
 
-    const { provider, owner, repo } = props;
+    const { provider, owner, repo, trustedRepositories } = props;
+
+    // Validate props
+    GithubActionsRole.validateProps(scope, props);
 
-    // Perform validations
-    GithubActionsRole.validateOwner(scope, owner);
-    GithubActionsRole.validateRepo(scope, repo);
+    // Unify the two ways of defining trusted repositories
+    const subjects = trustedRepositories || [{ owner: owner!, repo: repo!, filter: props.filter }];
+
+    // Perform validations on each trusted repository
+    subjects.forEach((subject) => {
+      GithubActionsRole.validateOwner(scope, subject.owner);
+      GithubActionsRole.validateRepo(scope, subject.repo);
+    });
 
-    // Prepare values
-    const subject = GithubActionsRole.formatSubject(props);
+    // Extract IAM Role props
     const roleProps = GithubActionsRole.extractRoleProps(props);
 
     // The actual IAM Role creation
     super(scope, id, {
       ...roleProps,
       assumedBy: new iam.WebIdentityPrincipal(provider.openIdConnectProviderArn, {
-        StringLike: {
+        'ForAnyValue:StringLike': {
```

Comment:
This should be safe even if just one property is passed. From the [docs](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_condition-single-vs-multi-valued-context-keys.html#reference_policies_condition-multi-valued-context-keys):

> ForAnyValue – This qualifier tests whether at least one member of the set of request context key values matches at least one member of the set of context key values in your policy condition. The context key returns true if any one of the context key values in the request matches any one of the context key values in the policy. For no matching context key or a null dataset, the condition returns false.

If we think people might be worried about seeing a `cdk diff` with this, we could optionally use `ForAnyValue` only when multiple repos are passed.